purchase_advance_payment/tests/test_purchase_advance_payment.py

```diff
@@ -106,8 +106,8 @@ def setUpClass(cls):
         cls.currency_usd = cls.env["res.currency"].search([("name", "=", "USD")])
         cls.currency_rate = cls.env["res.currency.rate"].create(
             {
-                "rate": 1.20,
-                "currency_id": cls.currency_usd.id,
+                "inverse_company_rate": 1.20,
+                "currency_id": cls.currency_euro.id,
```

Comment:
Is this really gonna help? Thing with tests that involve company currency or related is tricky for main company. Cause some modules can replace the currency, but if some other modules are installed it can change it to another. So you end up with either USD or EUR currency (but it is not consistent). This can fail if other case happens. Unless you are sure here we will always have EUR?

Reply:
Well, all previous tests are made assuming, that USD is base currency. My only change was made to fix the failing tests so  that rate is created for EUR currency and not USD. Which did not make any sense, honestly I don't know why it worked prior to v17.

Reply:
As I said, depending what module is installed, it can have different currency. So there are cases when company currency is different and then tests become inconsistent. For example tests when different COA (localization) is installed:). Better approach would be to test on fresh company, where we would not need to rely on set data that cant even be changed. Though for now, lets hope this won't blow up randomly, which is the most annoying failure:)

Reply:
I agree. Although since this is an OCA module, I'm surprised it didn't blow up in previous versions...